collect_app/src/main/java/org/odk/collect/android/utilities/DialogUtils.java

```diff
@@ -78,6 +83,45 @@ public static void showDialog(Dialog dialog, Activity activity) {
         }
     }
 
+    /**
+     * Creates a confirm/cancel dialog for deleting repeats.
+     */
+    public static void createDeleteRepeatConfirmDialog(Context context, Runnable onDeleted, Runnable onCanceled) {
+        FormController formController = Collect.getInstance().getFormController();
+        String name = formController.getLastRepeatedGroupName();
+        int repeatcount = formController.getLastRepeatedGroupRepeatCount();
+        if (repeatcount != -1) {
+            name += " (" + (repeatcount + 1) + ")";
```

Comment:
In the example where the repeat section is "Friends" and each instance of the repeated group is a friend, what does this show?  Suppose there are a total of three friends (Alice, Bob, Charlie) and the currently shown friend is Bob.  Will it show "Friends (3)" because there are three friends, or "Friends (2)" because Bob is #2, or "Bob (2)"?

Reply:
Friends (2):

![screenshot from 2019-01-09 12-32-21](https://user-images.githubusercontent.com/2047062/50917057-a4e1ca00-140a-11e9-8d1f-bf17d9183570.png)

But this is old logic, and I think it should be improved as you described in https://github.com/opendatakit/collect/pull/2762#issuecomment-451619959. There are probably numerous other places in the code where this type of string is built... we shoud do a quick audit if we want to update it.